README.md

```diff
@@ -251,6 +254,30 @@ Future.node(done => fs.readFile('package.json', 'utf8', done))
 //> "{...}"
 ```
 
+#### loop
+##### `.loop :: (b -> Future a (Iteration b)) -> b -> Future a b`
+
+Stack- and memory-safe asynchronous "recursion" based on [FantasyLand ChainRec][26].
+Calls the given function with the initial value, and expects a Future of an
+[Iteration](#type-signatures). If the Iteration is incomplete (`{done: false}`),
+then the function is called again with the Iteration value until it returns a
+Future of a complete (`{done: true}`) Iteration.
+
+```js
+const Next = x => ({done: false, value: x});
+const Done = x => ({done: true, value: x});
+Future.loop((x => Future.of(x < 1000000 ? Next(x + 1) : Done(x))), 0)
+.fork(console.error, console.log);
+//> 1000000
+```
+
+#### chainRec
+##### `.chainRec :: ((((b -> Next b), (b -> Done b), b) -> Future a (Iteration b)), b) -> Future a b`
```

Comment:
What do you think of the following?

```hs
chainRec :: (((b -> Next), (c -> Done), b) -> Future a Iteration) -> b -> Future a c
```

I believe it's correct, and I think it's also the most clear:

* It tells you which callback creates a Next, and which creates a Done.
* It shows that a Future of Iteration must be returned
* It doesn't go into the specifics of the Iteration type, leaving the signature still relatively short and legible.